content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-extensions.md

```diff
@@ -0,0 +1,48 @@
+description: Information on how to extend the terrain system to accomplish various goals.
+weight: 400
+---
+The terrain system is designed to be highly decoupled and easily extensible at a number of different layers.
+
+## Gradient Components
+
+Gradient components are the underlying data source for height data and surface weight data. New gradient component types can be created as-needed to provide data from different sources, such as streaming satellite data, or to modify existing gradient data, such as adding complex erosion filters.
+
+New gradient components need to support the following at a minimum:
+
+* Implement the `GradientRequestBus` to generically return a floating-point output value for an input world position.
+* Declare that it implements the `GradientService` in `GetProvidedServices`.
+* Support thread safety. Ideally, this will mean using a `shared_mutex` with a `shared_lock` in the query APIs and a `unique_lock` for any places that change the underlying data used in the query API methods. This will let multiple queries run in parallel while safely blocking on data changes.
+* Call `DependencyNotificationBus::Events::OnCompositionChanged` or `DependencyNotificationBus::Events::OnCompositionRegionChanged` whenever any data within the gradient changes. This notifies everything "above" the gradient that it will need to refresh itself.
+
+Ideally, new gradient components would also support the following:
+
+* Use the **Shape** components to define how the gradient data should map into world space.
+* Support the **Gradient Transform Modifier** component for performing arbitrary transforms to the gradient data.
+* By convention, all gradients currently listen to `OnEntityVisibilityChanged` and use that to enable or disable the gradient data in the Editor. This makes it easy to turn gradients on and off while authoring, though it's important to recognize that they will still be active at runtime regardless of the authoring visibility setting.
+
+## Terrain Components
+
+Each of the terrain components can be replaced with new components that meet the same requirements. For example, a new component could be created that streams down real-world satellite height data as a replacement for the **Terrain Height Gradient List**. It's also possible to combine components together, such as creating a simplified Terrain Spawner component that implements the functionality of the **Terrain Layer Spawner**, **Terrain Height Gradient List**, and **Terrain Surface Gradient List** on one single component.
+
+Any replacement components need to support the following:
+
+* Implement the request bus that matches the component being replaced (`TerrainSpawnerRequestBus`, `TerrainAreaHeightRequestBus`, `TerrainAreaSurfaceRequestBus`, etc).
+* Declare that it implements each service that matches the component being replaced (`TerrainAreaService`, `TerrainHeightProviderService`, `TerrainSurfaceProviderService`, etc) in `GetProvidedServices`.
```

Comment:
- Nit, this is mostly personal preference and standard docs style. I do prefer "For example", or similar, as makes it clear what the info your listing is for. 
  Either way is okay for me, but I strongly suggest moving the examples in the second bullet to the end of the sentence. Since the examples have long names, it interrupts the sentence. 
- Also, for lists, keep your punctuation consistent. Either no period or a period for all of them. 
```suggestion
* Implement the request bus that matches the component being replaced. For example, `TerrainSpawnerRequestBus`, `TerrainAreaHeightRequestBus`, `TerrainAreaSurfaceRequestBus`, etc.
* Declare that it implements each service that matches the component being replaced in `GetProvidedServices`. For example, TerrainAreaService`, `TerrainHeightProviderService`, `TerrainSurfaceProviderService`, etc.
```

Reply:
We're also not supposed to use Latin abbreviations, so how's this instead?

```suggestion
* Implement the request bus that matches the component being replaced. For example, `TerrainSpawnerRequestBus`, `TerrainAreaHeightRequestBus`, or `TerrainAreaSurfaceRequestBus`.
* Declare that it implements each service that matches the component being replaced in `GetProvidedServices`. For example, `TerrainAreaService`, `TerrainHeightProviderService`, or `TerrainSurfaceProviderService`.
```

Reply:
You're right, we usually don't use latin abbrevs. I know that with this going into the Engine Developer Guide eventually, that will be under a Style Guide that's decided between the technical sigs and sig-docs-community. So I've been reviewing it more lightly on some rules. Not using latin abbrevs can be a style we choose to keep and I'm +1 for that.